Sources/PaymentsCore/Networking/APIClient.swift

```diff
@@ -3,15 +3,24 @@ import Foundation
 public final class APIClient {
     public typealias CorrelationID = String
 
-    public var urlSession: URLSession
-    public var environment: Environment
+    private var urlSession: URLSessionProtocol
+    private var environment: Environment
 
-    private let decoder = JSONDecoder()
+    private var decoder: JSONDecoder {
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+        return decoder
+    }
```

Comment:
Instead of being computed, this should be an instance var - 
```suggestion
    private var decoder: JSONDecoder = {
        let decoder = JSONDecoder()
        decoder.keyDecodingStrategy = .convertFromSnakeCase
        return decoder
    }()
```

Reply:
Good call - will update!